dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/client/HarvestingClientsPage.java

```diff
@@ -451,30 +419,30 @@ public void saveClient(ActionEvent ae) {
             harvestingClient.setScheduled(false);
         }
 
-        // will try to save it now:
 
-        try {
-            harvestingClient = engineService.submit(new UpdateHarvestingClientCommand(dvRequestService.getDataverseRequest(), harvestingClient));
+        HarvestingClient finalHarvestingClient = harvestingClient;
+        Try<HarvestingClient> harvestingClientOperation = Try.of(() -> harvestingClientsService.updateHarvestingClient(finalHarvestingClient));
 
+        if(harvestingClientOperation.isSuccess()) {
+            harvestingClient = harvestingClientOperation.get();
             configuredHarvestingClients = harvestingClientService.getAllHarvestingClients();
-
             if (!harvestingClient.isScheduled()) {
                 dataverseTimerService.removeHarvestTimer(harvestingClient);
             }
-            JsfHelper.addFlashSuccessMessage(BundleUtil.getStringFromBundle("harvest.update.success") + harvestingClient.getName());
-
-        } catch (CommandException ex) {
-            logger.log(Level.WARNING, "Failed to save harvesting client", ex);
-            JsfHelper.JH.addMessage(FacesMessage.SEVERITY_ERROR,
-                                    BundleUtil.getStringFromBundle("harvest.save.failure1"),
-                                    ex.getMessage());
-        } catch (Exception ex) {
-            JH.addMessage(FacesMessage.SEVERITY_FATAL, BundleUtil.getStringFromBundle("harvest.save.failure2"));
-            logger.log(Level.SEVERE, "Failed to save harvesting client (reason unknown)." + ex.getMessage(), ex);
+            JsfHelper.addFlashSuccessMessage(BundleUtil.getStringFromBundle("harvest.update.success", harvestingClient.getName()));
+        } else {
```

Comment:
Same as above Imo this could be in OnFailure with private method, there is just code noise that we don't really care about reading this method since failure handling is always the same.